### PR TITLE
[skip ci] PR engagement rules

### PR DESCRIPTION
**Pull request by troydai**
I write up this to purpose some guidelines for how PR should be composed and when will we begin reviewing PR. Please let know your thought.

This is just a draft as of now.

### CODE REVIEW

**Comment by troydai** [comment]
@mayurid

File: doc/pull_request_engagement_rule.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# Pull Request Engagement Rules
+
+## Why do we need these rules
+
+We have 20 PR (pull request) opened on an average day, more when it is closer to final days of a sprint. Reviewing pull request cost time and energy. Automation jobs triggered by the PR also constantly compete for the finite resources.
+
+To better facilitate the reviewing process to ensure PR are reviewed and closed in timely fashion the engineering team set up the engagement rules to define when and how we will begin review pull requests.
```
Comment (williexu) [inline doc/pull_request_engagement_rule.md]: `To better facilitate the reviewing process to ensure PRs are reviewed and closed in a timely fashion, the engineering team has set up the following engagement rules to define when and how we will begin reviewing pull requests.`

File: doc/pull_request_engagement_rule.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+## What to do when open a PR
```
Comment (williexu) [inline doc/pull_request_engagement_rule.md]: `## What to do when opening a PR`

File: doc/pull_request_engagement_rule.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+### Pick a good subject
+
+Make the subject line clear and concise. Make it clear what area this PR indents to change.
```
Comment (williexu) [inline doc/pull_request_engagement_rule.md]: `intends`

File: doc/pull_request_engagement_rule.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+Bad examples:
+
+    * fix the wrong help
+    * Update _help.py
+    * not all VMSS autoscale settings has notifications, we need to verify …
+
+### More information in the PR body
+
+In additional to the subject line, following details should be specified in the PR body:
```
Comment (williexu) [inline doc/pull_request_engagement_rule.md]: In additional to the subject line, **the** following details

File: doc/pull_request_engagement_rule.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+* [Required] The milestone: Make it clear what sprint you expect to have this change shipped. We will prioritize the PR for nearest milestone.
+* [Required] The associcated issues.
+* [Optional] The area / command module this PR is intended for.
+* [Optional] Concerns
+
+### Skip CI
+
+If the PR only touches documentation, add `[skip ci]` to the subject line. [Reference](https://docs.travis-ci.com/user/customizing-the-build/#Skipping-a-build)
+
+## Prerequisit of code review
```
Comment (williexu) [inline doc/pull_request_engagement_rule.md]: `Prerequisites for a code review`

File: doc/pull_request_engagement_rule.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+* The CI must pass
+* The PR is for current sprint (unless there is no open PR in current sprint)
```
Comment (williexu) [inline doc/pull_request_engagement_rule.md]: **the** current sprint

**Pull request closed without merging**